modules/portworx/examples/simple/variables.tf

```
@@ -0,0 +1,45 @@
+# TEMPLATE: This file was automatically generated with `generate_addon_structure.sh`
+# TEMPLATE: and should be modified as necessary
+# TEMPLATE:
+# TEMPLATE: All variables must have a description and should declare their type.
+# TEMPLATE: Set defaults whenever possible but do not set defaults for required properties.
+# TEMPLATE: Declare all variables in this file, sprawling declarations are difficult to identify.
+# TEMPLATE:
+# TEMPLATE: https://www.terraform.io/docs/language/values/variables.html
+# TEMPLATE: https://www.terraform.io/docs/language/expressions/types.html
+#
+variable "metal_auth_token" {
+  type        = string
+  description = "Equinix Metal API key. Leave it empty to use your METAL_AUTH_TOKEN / TF_VAR_metal_auth_token environment variable"
+  sensitive   = true
+}
+
+variable "node_names" {
+  type        = list(string)
+  description = "Baremetal hostnames"
+  default     = null
+}
+
+variable "private_key_path" {
+  type        = string
+  description = "Path to an SSH key to use for the connection"
+  default     = null
+}
+
+variable "node_ips" {
+  type        = list(string)
+  description = "Baremetal ipv4 addresses"
+  default     = null
+}
+
+variable "node_user" {
+  type        = list(string)
+  description = "Baremetal username to login"
+  default     = null
+}
```

Use a single variable to ensure order and relation between hosts and their ssh config
```suggestion
variable "worker_nodes" {
  type = list(map(string))
  description = "List of Baremetal worker nodes: [ {ssh-user, private-key-path, ip} ], e.g. [ {\"root\", \"worke-01\", \"140.30.2.1\"} ]"
}
```